sdk/src/main/scala/com/horizen/SidechainState.scala

```diff
@@ -209,6 +211,17 @@ class SidechainState private[horizen] (stateStorage: SidechainStateStorage,
     if (allInputBoxesIds.size != allInputBoxesIds.toSet.size) {
       throw new IllegalArgumentException(s"Block ${mod.id} contains duplicated input boxes to open")
     }
+
+    //Check that we don't have multiple openStake transactions with the same forgerList index
+    val forgerListIndexes = new JArrayList[Int]()
+    mod.transactions.foreach(tx => {
+      if (tx.isInstanceOf[OpenStakeTransaction]) {
+        val openStakeTransaction = tx.asInstanceOf[OpenStakeTransaction]
```

Comment:
Maybe we should check the condition `if (!params.restrictForgers)` also here, even if it is checked by the validate(tx) method which is called afterwards

Reply:
Done

Reply:
So you are suggesting to remove the check `(!params.restrictForgers)` but keep the check on duplicated forgerIndex?